### PR TITLE
Checkout: update mobile submit button

### LINKED ISSUE #43954: Checkout: terms of service not always visible on mobile
**Issue:**
On mobile devices, when the final step is active, the submit button is fixed to the bottom of the screen. If there are more than 1 or 2 payment methods, it's possible for a user to be able to checkout while the terms of services are off-screen.

**Suggested Fix:**
Add a relative link to the ToS when the submit button is active and fixed-positioned.

### PR DESCRIPTION
**Pull request by michaeldcain**
This adds a relative link to the Terms of Service when the submit button is active and `position: fixed`. It also cleans up two small visual bugs where the fixed submit button overlaps the "total" in the final step, and the ApplePay button wasn't the correct height.

Fixes #43954 

Before | After
------------ | -------------
before | after

**To Test:**
- visit Checkout (`?flags=composite-checkout-force`) on a mobile device (or using the web inspector)
- complete the billing/contact information step if it's not already complete
- when the payment method step is active, the submit button should become `position: fixed`
- verify that the fixed submit button has a link to the Terms of Service
- click the link and verify that the Terms of Service are visible

### CODE REVIEW

File: packages/composite-checkout/src/components/checkout-steps.js [hunk 1]
```diff
@@ -799,7 +821,7 @@ function getStepNumberFromUrl() {
 		const stepNumber = parts.length > 1 ? parts[ 1 ] : 1;
 		return parseInt( stepNumber, 10 );
 	}
-	return 1;
+	return null;
```
Comment (sirbrillig) [inline packages/composite-checkout/src/components/checkout-steps.js:824]: Without this returning `1`, is there a way to get to the first step without having to add `#step1` on the hash?

The scenario I'm thinking of is:

- User visits checkout without autocomplete of contact details.
- User fills out contact details and presses Continue to go to step 2.
- User pressed Back in the browser; the URL changes to remove `#step2`.
- `useChangeStepNumberForUrl` triggers a hashchange and calls this function to figure out the new step.

With this change, I _think_ it means that the current step will remain at step 2.

Comment (michaeldcain) [reply packages/composite-checkout/src/components/checkout-steps.js]: I was able to reproduce the scenario above, so I returned the hash navigation to its old state and used `scrollIntoView` instead.

File: client/my-sites/checkout/composite-checkout/wpcom/components/wp-checkout.js [hunk 2]
```diff
@@ -516,3 +516,47 @@ function useUpdateCartLocationWhenPaymentMethodChanges(
...
 }
+
+function SubmitButtonHeader() {
+	const translate = useTranslate();
+
+	const scrollToTOS = () => document.getElementById( 'checkout-terms' ).scrollIntoView();
+
+	return (
+		<SubmitButtonHeaderUI>
+			{ translate( 'By continuing, you agree to our {{button}}Terms of Service{{/button}}.', {
```
Comment (a8ci18n) [inline client/my-sites/checkout/composite-checkout/wpcom/components/wp-checkout.js:527]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **25** times:
`translate( 'By continuing, you agree to our {{tosLink}}Terms of Service{{/tosLink}}.' )` *ES Score: 6*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/checkout-submit-button)

**Comment by sirbrillig** [comment]
I'm seeing a small regression where at desktop width the "pay" button is not as wide as it should be.

Screen Shot 2020-07-08 at 5 42 11 PM

Screen Shot 2020-07-08 at 5 41 19 PM

**Comment by michaeldcain** [comment]
@sirbrillig good catch on the full-width button. I fixed the broken selector.

**Pull request merged**